pypeit/core/coadd.py

```diff
@@ -605,120 +607,159 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     if np.array_equal(wave_new, wave_old):
-        return flux_old, ivar_old, mask_old
+        return flux_old, ivar_old, gpm_old
 
-    # make the mask array to be float, used for interpolation
-    masks_float = mask_old.astype(float)
-    wave_mask = wave_old > 1.0 # Deal with the zero wavelengths
+    wave_gpm = wave_old > 1.0 # Deal with the zero wavelengths
     if sensfunc:
-        delta_wave_interp = wvutils.get_delta_wave(wave_old, wave_mask)
-        flux_interp = flux_old[wave_mask]/delta_wave_interp[wave_mask]
-        ivar_interp = ivar_old[wave_mask]*delta_wave_interp[wave_mask]**2
+        delta_wave_interp = wvutils.get_delta_wave(wave_old, wave_gpm)
+        flux_interp = flux_old[wave_gpm]/delta_wave_interp[wave_gpm]
+        ivar_interp = ivar_old[wave_gpm]*delta_wave_interp[wave_gpm]**2
     else:
-        flux_interp = flux_old[wave_mask]
-        ivar_interp = ivar_old[wave_mask]
+        flux_interp = flux_old[wave_gpm]
+        ivar_interp = ivar_old[wave_gpm]
 
-    flux_new = scipy.interpolate.interp1d(wave_old[wave_mask], flux_interp, kind='cubic',
+    flux_new = scipy.interpolate.interp1d(wave_old[wave_gpm], flux_interp, kind='cubic',
                                     bounds_error=False, fill_value=np.nan)(wave_new)
-    ivar_new = scipy.interpolate.interp1d(wave_old[wave_mask], ivar_interp, kind='cubic',
+    ivar_new = scipy.interpolate.interp1d(wave_old[wave_gpm], ivar_interp, kind='cubic',
                                     bounds_error=False, fill_value=np.nan)(wave_new)
-    mask_new_tmp = scipy.interpolate.interp1d(wave_old[wave_mask], masks_float[wave_mask], kind='cubic',
-                                        bounds_error=False, fill_value=np.nan)(wave_new)
-    # Don't allow the ivar to be every less than zero
+    # Interpolate a floating-point version of the mask
+    gpm_new_tmp = scipy.interpolate.interp1d(wave_old[wave_gpm], gpm_old.astype(float)[wave_gpm],
+                                             kind='cubic', bounds_error=False,
+                                             fill_value=np.nan)(wave_new)
+    # Don't allow the ivar to be ever be less than zero
     ivar_new = (ivar_new > 0.0)*ivar_new
-    mask_new = (mask_new_tmp > 0.8) & (ivar_new > 0.0) & np.isfinite(flux_new) & np.isfinite(ivar_new)
-    return flux_new, ivar_new, mask_new
+    gpm_new = (gpm_new_tmp > 0.8) & (ivar_new > 0.0) & np.isfinite(flux_new) & np.isfinite(ivar_new)
+    return flux_new, ivar_new, gpm_new
 
-def interp_spec(wave_new, waves, fluxes, ivars, masks, sensfunc=False):
+
+# TODO: ``sensfunc`` should be something like "conserve_flux". It would be
+# useful to compare these resampling routines against
+# `pypeit.sampling.Resample`.
+def interp_spec(wave_new, waves, fluxes, ivars, gpms, sensfunc=False):
     """
-    Utility routine to interpolate a set of spectra onto a new
-    wavelength grid, wave_new
+    Interpolate a set of spectra onto a new wavelength grid.
```

Comment:
Please do not suppress the shapes of the output arrays that were written in the original documentation. I appreciate your improvements to the docs, butI'm trying to follow the scikit-learn documentation convention of always specifying the shapes of arrays.  Your updated documentation no longer makes it explicit which dimension is the number of spectral pixels and which dimension is the number of images.

Reply:
Done.